---
title: How to use pipe with R
date: '2021-12-21'
slug: use pipe with R
---

# Set work directory and loading packages

```
setwd('E:/thermokast_lakes/water_microbes/')
PKGs <- c('ape', 'Biostrings', 'phyloseq',               
              'tidyr', 'dplyr', 'ggplot2')
lapply(PKGs, require, character.only = TRUE, warn.conflicts = FALSE)
```


# Conduct a phyloseq project
```
#read in metadata
metadata <- read.csv("./meta_analysis/data/meta_data/sample_data.csv",
                     header = T, row.names = 1)

#read in otu table
meta.otu.table <- read.csv("./meta_analysis/data/meta_data/meta_otu_table.csv",
                           header = T, row.names = 1, stringsAsFactors = F)
meta.otu.table <- as.matrix(meta.otu.table)

#read in taxonomy
meta.taxonomy <- read.csv("./meta_analysis/data/meta_data/meta_taxonomy.csv",sep=",",row.names=1)
meta.taxonomy <- as.matrix(meta.taxonomy)

# read in tree
meta.phy.tree <- read_tree("./meta_analysis/data/meta_data/meta_tree.nwk")

#read in represent dna sequences
meta.ref.seqs <- readDNAStringSet(file = "./meta_analysis/data/meta_data/meta_ref_seqs.fasta",
                                  format = "fasta", nrec = -1L, skip = 0L, seek.first.rec = FALSE, use.names = TRUE)
meta.otu.table <- otu_table(meta.otu.table, taxa_are_rows = TRUE)
meta.tax.table <- tax_table(meta.taxonomy)
meta.table <- sample_data(metadata)
meta_physeq <- phyloseq(meta.tax.table, meta.otu.table, meta.table, meta.phy.tree, meta.ref.seqs)
meta_physeq
```


    phyloseq-class experiment-level object
    otu_table()   OTU Table:         [ 10250 taxa and 320 samples ]
    sample_data() Sample Data:       [ 320 samples by 38 sample variables ]
    tax_table()   Taxonomy Table:    [ 10250 taxa by 8 taxonomic ranks ]
    phy_tree()    Phylogenetic Tree: [ 10250 tips and 10227 internal nodes ]
    refseq()      DNAStringSet:      [ 10250 reference sequences ]


# Calculate the diversity
```
diversity <- estimate_richness(meta_physeq, measures = c("Chao1", 'Shannon', 'Simpson'))
meta_diversity <- cbind(diversity, Region = metadata$Region,
                        Site = metadata$Site,  Sitegroup1 = metadata$Sitegroup1)
```


```
meta_diversity[1:5, 1:7]
```


<table class="dataframe">
<caption>A data.frame: 5 × 7</caption>
<thead>
	<tr><th></th><th scope=col>Chao1</th><th scope=col>se.chao1</th><th scope=col>Shannon</th><th scope=col>Simpson</th><th scope=col>Region</th><th scope=col>Site</th><th scope=col>Sitegroup1</th></tr>
	<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>ERR4650773</th><td> 44.0000</td><td> 9.625787</td><td>1.661566</td><td>0.742798</td><td>Pan-Arctic</td><td>A</td><td>North_East Siberia</td></tr>
	<tr><th scope=row>ERR4650774</th><td> 45.1000</td><td> 6.481545</td><td>1.660527</td><td>0.726742</td><td>Pan-Arctic</td><td>A</td><td>North_East Siberia</td></tr>
	<tr><th scope=row>ERR4650775</th><td>127.2308</td><td>23.000918</td><td>2.428847</td><td>0.802124</td><td>Pan-Arctic</td><td>A</td><td>North_East Siberia</td></tr>
	<tr><th scope=row>ERR4650776</th><td>121.1765</td><td>17.164433</td><td>2.619853</td><td>0.841466</td><td>Pan-Arctic</td><td>A</td><td>North_East Siberia</td></tr>
	<tr><th scope=row>ERR4650777</th><td>105.5652</td><td>10.045071</td><td>2.966866</td><td>0.900248</td><td>Pan-Arctic</td><td>A</td><td>North_East Siberia</td></tr>
</tbody>
</table>



# Manipulating and analyzing data with dplyr and tidyr

```
tp_alpha_div <-  meta_diversity %>% 
  filter(Region == 'Tibetan Plateau') %>%
  dplyr::select(-c(2)) %>%
  group_by(Region, Site, Sitegroup1) %>%
  dplyr::summarise(across(, mean, na.rm = TRUE)) %>%
  tidyr::gather(diversity, value, -c('Region', 'Site', 'Sitegroup1')) %>%
  dplyr::mutate(Sitegroup1 = factor(Sitegroup1, levels = c('MD', 'QL', 'QZR')))
tp_alpha_div[1:6, ]
```

    `summarise()` regrouping output by 'Region', 'Site' (override with `.groups` argument)


​    


<table class="dataframe">
<caption>A grouped_df: 6 × 5</caption>
<thead>
	<tr><th scope=col>Region</th><th scope=col>Site</th><th scope=col>Sitegroup1</th><th scope=col>diversity</th><th scope=col>value</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>Tibetan Plateau</td><td>Z1 </td><td>QL </td><td>Chao1</td><td>496.7214</td></tr>
	<tr><td>Tibetan Plateau</td><td>Z10</td><td>QZR</td><td>Chao1</td><td>415.7837</td></tr>
	<tr><td>Tibetan Plateau</td><td>Z11</td><td>QZR</td><td>Chao1</td><td>376.7769</td></tr>
	<tr><td>Tibetan Plateau</td><td>Z12</td><td>QZR</td><td>Chao1</td><td>387.1121</td></tr>
	<tr><td>Tibetan Plateau</td><td>Z13</td><td>QZR</td><td>Chao1</td><td>227.1032</td></tr>
	<tr><td>Tibetan Plateau</td><td>Z14</td><td>QZR</td><td>Chao1</td><td>517.1068</td></tr>
</tbody>
</table>



# Plot with ggplot2
```
alpha_region_plot <- ggplot(tp_alpha_div, aes(x = Sitegroup1, y = value, fill = Sitegroup1)) +
  geom_violin(trim=T, width=0.5, aes(fill = Sitegroup1), colour = "#000000") +
  scale_fill_manual(values= c('#d95f02', '#1b9e77', 'red')) +
  geom_boxplot(width=0.1, fill="white", colour = "#000000") +
  labs(x = 'Sitegroup1', y = 'Alpha diversity') +
  facet_wrap(~diversity, scales = 'free') +
  theme_bw() +
  theme(strip.text = element_text(size = 14),
        axis.line = element_line(colour = "black"),
        axis.title = element_text(size = 16, colour = 'black'),
        axis.text = element_text(size = 14, colour = 'black'),
        axis.text.x = element_text(colour='black', size = 14, 
                                   angle = 45, hjust = 1),
        panel.grid = element_blank(), legend.position = 'none')
options(repr.plot.width = 17, repr.plot.height = 8.3, repr.plot.res = 300)
print(alpha_region_plot)
```


​    
![](/images/output_5_0.png)
​    

